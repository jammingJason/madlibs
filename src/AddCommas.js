import React, { useState } from 'react';
const AddCommas = () => {
  const handleChange = (evt) => {
    const { value } = evt.target;
    addCommas(value);
  };

  const addCommas = (num) => {
    const newNum = document.getElementById('numCommas');
    if (num < 0) {
      let anotherNum = num.toString().split('-');
      let newString = insertCommas(anotherNum[1]);
      newNum.innerText = '-' + newString;
      return;
    }
    if (num.indexOf('.') === -1) {
      const newString = insertCommas(num);
      newNum.innerText = newString;
    } else {
      let newNumber = num.split('.');
      const newString = insertCommas(newNumber[0]);
      newNum.innerText = newString + '.' + newNumber[1];
    }
  };

  const insertCommas = (num) => {
    console.log('Raw number being passed ==>' + num);
    const numArray = [...num];
    let newArray = [];
    let count = 0;

    if (numArray.length >= 4) {
      for (let i = numArray.length - 1; i >= 0; i--) {
        if (count === 3) {
          newArray.unshift(',');
          count = 0;
        }
        newArray.unshift(numArray[i]);
        //   console.log(count);
        count++;
      }
    } else {
      newArray = [...numArray];
    }

    let newString = '';
    newArray.map((nums) => {
      newString = newString + nums;
    });
    return newString;
  };

  return (
    <>
      <form>
        <label htmlFor="number">Number : </label>
        <input
          id="numberHolder"
          className="number"
          key={1}
          type="text"
          name="number"
          placeholder="12345"
          onChange={handleChange}
          role="numberHolder"
        />

        <div>
          <label id="numCommas">New Number Goes Here</label>
        </div>
      </form>
    </>
  );
};
export default AddCommas;
