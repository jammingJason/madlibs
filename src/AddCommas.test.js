// const addCommas = require("./addCommas");
import { render, screen, fireEvent } from '@testing-library/react';
import AddCommas from './AddCommas';

describe('#AddCommas', () => {
  test('it is a function', () => {
    expect(typeof AddCommas).toBe('function');
  });
});

// smoke test
it('renders without crashing', function () {
  render(<AddCommas />);
});

// snapshot test
it('matches snapshot', function () {
  const { asFragment } = render(<AddCommas />);
  expect(asFragment()).toMatchSnapshot();
});

it('handles comma insertions for positive numbers', function () {
  const { getByText, getByRole } = render(<AddCommas />);
  const inputNode = screen.getByPlaceholderText('12345');
  const h2 = getByText('New Number Goes Here');

  fireEvent.change(inputNode, { target: { value: '2356' } });

  expect(h2.innerText).toBe('2,356');
});

it('handles comma insertions for negative numbers', function () {
  const { getByText, getByRole } = render(<AddCommas />);
  const inputNode = screen.getByPlaceholderText('12345');
  const h2 = getByText('New Number Goes Here');

  fireEvent.change(inputNode, { target: { value: '-2356' } });

  expect(h2.innerText).toBe('-2,356');
});

it('handles comma insertions for decimals', function () {
  const { getByText, getByRole } = render(<AddCommas />);
  const inputNode = screen.getByPlaceholderText('12345');
  const h2 = getByText('New Number Goes Here');

  fireEvent.change(inputNode, { target: { value: '-2356.0253' } });

  expect(h2.innerText).toBe('-2,356.0253');
});
