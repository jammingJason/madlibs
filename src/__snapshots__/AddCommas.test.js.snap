// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="number"
    >
      Number : 
    </label>
    <input
      class="number"
      id="numberHolder"
      name="number"
      placeholder="12345"
      role="numberHolder"
      type="text"
    />
    <div>
      <label
        id="numCommas"
      >
        New Number Goes Here
      </label>
    </div>
  </form>
</DocumentFragment>
`;
